validations/src/validations.rs

```diff
@@ -986,20 +994,20 @@ pub fn validate_tally_transaction<'a>(
             }
         } else {
             // Validation of each rewarded address was a revealer
-            if dr_state.info.reveals.get(&output.pkh).is_none() {
+            if reveals_count > 0 && dr_state.info.reveals.get(&output.pkh).is_none() {
                 return Err(TransactionError::RevealNotFound.into());
             }
 
             // Validation of each rewarded address was not a liar
-            if slashed_hs.contains(&output.pkh) {
+            if reveals_count > 0 && slashed_hs.contains(&output.pkh) {
```

Comment:
The expression evaluated first in both conditionals is the same. I'd favor nested conditionals:

```rust
if reveals_count > 0 {
    // Make sure every rewarded address is a revealer
    if dr_state.info.reveals.get(&output.pkh).is_none() {
        return Err(TransactionError::RevealNotFound.into());
    }
    // Make sure every rewarded address passed the tally function, a.k.a. "is honest" / "is not a liar"
    if slashed_hs.contains(&output.pkh) {
        return Err(TransactionError::DishonestReward.into());
    }
}
```